lib/spack/spack/test/installer.py

```diff
@@ -447,8 +448,12 @@ def test_packages_needed_to_bootstrap_compiler_none(install_mockery):
     assert not packages
 
 
-def test_packages_needed_to_bootstrap_compiler_packages(install_mockery,
-                                                        monkeypatch):
+@pytest.mark.xfail(
+    reason="fails when assuming Spec.package can only be called on concrete specs"
+)
+def test_packages_needed_to_bootstrap_compiler_packages(
```

Comment:
It is my understanding that the "install_missing_compilers" feature is not working with `clingo`, so I'm not sure it's worth checking how to make this test pass. Note that it disables any kind of concretization at the moment.

Reply:
@trws @tgamblin :point_up: